src/services/LogFileManager/index.ts

```diff
@@ -202,21 +205,40 @@ class LogFileManager {
         fileData: Uint8Array,
         decoderOptions: DecoderOptions
     ): Promise<Decoder> {
-        let decoder: Decoder;
-        if (fileName.endsWith(".jsonl")) {
-            decoder = new JsonlDecoder(fileData, decoderOptions);
-        } else if (fileName.endsWith(".clp.zst")) {
-            decoder = await ClpIrDecoder.create(fileData, decoderOptions);
-        } else {
-            throw new Error(`No decoder supports ${fileName}`);
-        }
-
         if (fileData.length > MAX_V8_STRING_LENGTH) {
             throw new Error(`Cannot handle files larger than ${
                 formatSizeInBytes(MAX_V8_STRING_LENGTH)
             } due to a limitation in Chromium-based browsers.`);
         }
 
+        let decoder: Decoder;
+
+        const checkFileExtension = async () => {
+            if (fileName.endsWith(".jsonl")) {
+                decoder = new JsonlDecoder(fileData, decoderOptions);
+            } else if (fileName.endsWith(".clp.zst")) {
+                decoder = await ClpIrDecoder.create(fileData, decoderOptions);
+            } else {
+                throw new Error(`No decoder supports ${fileName}`);
+            }
+        };
+
+        if (fileData.length < MAGIC_NUMBER_SIZE) {
+            await checkFileExtension();
+
+            return decoder;
+        }
+        const magicNumber = (fileData[0] << BYTE_SIZE * 3) |
+            (fileData[1] << BYTE_SIZE * 2) | (fileData[2] << BYTE_SIZE) | fileData[3];
```

Comment:
This will only work for formats that specifies 4-byte magic numbers, right? Can we make it work for 2-byte / a variable number of bytes of magic numbers as well?

For example, the gzip format can be identified by magic numbers `1F 8B`. If the algorithm is "deflate", we can expect `1F 8B 08`, which is still only 3 bytes.